process/transaction/interceptedTransaction.go

```diff
@@ -165,12 +186,49 @@ func (inTx *InterceptedTransaction) CheckValidity() error {
 			return err
 		}
 
+		err = inTx.verifyIfRelayedTxV2(inTx.tx)
+		if err != nil {
+			return err
+		}
+
 		inTx.whiteListerVerifiedTxs.Add([][]byte{inTx.Hash()})
 	}
 
 	return nil
 }
 
+func (inTx *InterceptedTransaction) verifyIfRelayedTxV2(tx *transaction.Transaction) error {
+	funcName, userTxArgs, err := inTx.argsParser.ParseCallData(string(tx.Data))
+	if err != nil {
+		return nil
+	}
+	if core.RelayedTransactionV2 != funcName {
+		return nil
+	}
+
+	userTx, err := createRelayedV2(tx, userTxArgs)
+	if err != nil {
+		return err
+	}
+
```

Comment:
We might need the enable epoch flag check here if we run into incompatibility issues.

Reply:
yes. I was tough about that too. but intercepted transaction is not an object which persist over time. thus needs to get the activation from some place. Still thinking whether this is a must or not.

Reply:
yup, will keep this as a theoretical problem for now.